tests/InstrEngineTests/NaglerInstrumentationMethod/NaglerInstrumentationMethod.cpp

```diff
@@ -1187,6 +1202,133 @@ HRESULT CInstrumentationMethod::PerformSingleReturnInstrumentation(IMethodInfo*
     return S_OK;
 }
 
+HRESULT CInstrumentationMethod::AddExceptionHandler(IMethodInfo* pMethodInfo, IInstructionGraph* pInstructionGraph)
+{
+    HRESULT hr;
+
+    IModuleInfo* pModuleInfo;
+    IfFailRet(pMethodInfo->GetModuleInfo(&pModuleInfo));
+
+    IType* pReturnType;
+    IfFailRet(pMethodInfo->GetReturnType(&pReturnType));
+
+    CorElementType returnCorElementType;
+    IfFailRet(pReturnType->GetCorElementType(&returnCorElementType));
+
+    CComPtr<ILocalVariableCollection> localVars;
+    IfFailRet(pMethodInfo->GetLocalVariables(&localVars));
+
+    DWORD returnIndex = -1;
+    if (returnCorElementType != ELEMENT_TYPE_VOID)
+    {
+        IfFailRet(localVars->AddLocal(pReturnType, &returnIndex));
+    }
+
+    CComPtr<IMetaDataImport> spIMetaDataImport;
+    IfFailRet(pModuleInfo->GetMetaDataImport(reinterpret_cast<IUnknown**>(&spIMetaDataImport)));
+
+    CComPtr<IMetaDataAssemblyImport> spIMetaDataAssemblyImport;
+    IfFailRet(pModuleInfo->GetMetaDataImport(reinterpret_cast<IUnknown**>(&spIMetaDataAssemblyImport)));
+
+    mdAssembly resolutionScope;
+    spIMetaDataAssemblyImport->GetAssemblyFromScope(&resolutionScope);
+
+    std::wstring strExceptionTypeName(_T("System.Exception"));
+
+    ULONG chName = MAX_PATH;
+    std::wstring strName(chName, wchar_t());
+    DWORD cTokens;
+    mdTypeRef currentTypeRef;
+    mdTypeRef targetTypeRef = mdTypeRefNil;
+    MicrosoftInstrumentationEngine::CMetadataEnumCloser<IMetaDataImport> spHEnSourceTypeRefs(spIMetaDataImport, nullptr);
+    while (SUCCEEDED(spIMetaDataImport->EnumTypeRefs(spHEnSourceTypeRefs.Get(), &currentTypeRef, 1, &cTokens)) && cTokens > 0)
+    {
+        spIMetaDataImport->GetTypeRefProps(currentTypeRef, &resolutionScope, &strName[0], static_cast<ULONG>(strName.size()), &chName);
+        if (0 == wcscmp(strName.c_str(), strExceptionTypeName.c_str()))
+        {
+            targetTypeRef = currentTypeRef;
+            break;
+        }
+    }
+
+    if (targetTypeRef == mdTypeRefNil)
+    {
+        return E_FAIL;
+    }
+
+
+    CComPtr<IInstructionFactory> pInstructionFactory;
+    IfFailRet(pMethodInfo->GetInstructionFactory(&pInstructionFactory));
+
+    CComPtr<IInstruction> sptrExitLabel;
```

Comment:
>sptr [](start = 26, length = 4)

nit: our coding style doesn't typically mark pointers as "smart" we just use pVariableName.

Reply:
I think I copy pasted this prefix from here: https://github.com/microsoft/CLRInstrumentationEngine/blob/develop/tests/InstrEngineTests/NaglerInstrumentationMethod/NaglerInstrumentationMethod.cpp#L1009

But I mix several prefix styles in the same method, so I've tried to standardize on a single "p" in this commit: https://github.com/robertpi/CLRInstrumentationEngine/commit/a22fc6c204007911e9e2ff5fcfac6261382369df